File: python/buggy_code.py
#!/usr/bin/env python3
# Buggy Python code for testing

# Missing colon
def main()
    print("Starting buggy code")
    
    # Indentation error
    x = 10
      y = 20  # Wrong indentation
    
    # Undefined variable
    print(z)
    
    # Missing closing parenthesis
    print("Hello"
    
    # Wrong operator
    if x = 10:  # Should be ==
        print("x is 10")
    
    # TypeError
    numbers = [1, 2, 3]
    numbers + "4"
    
    # Missing self parameter
    class Test:
        def method():  # Missing self
            return "test"
    
    # Infinite recursion (no base case)
    def factorial(n):
        return n * factorial(n-1)
    
    # Calling method before definition
    my_func()
    
    def my_func():
        return "defined later"

if __name__ == "__main__":
    main(

File: java/BuggyCode.java
public class BuggyCode {
    public static void main(String[] args) {
        // Missing semicolon below
        System.out.println("Starting buggy code")
        
        // Variable not declared
        x = 10;
        
        // Missing closing brace
        for (int i = 0; i < 5; i++ {
            System.out.println("Count: " + i);
        
        // Unclosed string
        String message = "Hello world;
        
        // Method not defined
        undefinedMethod();
        
        // Wrong array syntax
        int[] numbers = new int[5];
        numbers[5] = 10; // Index out of bounds
        
        // Missing parenthesis
        if (x > 5 {
            System.out.println("x is greater than 5");
        }
    }
    
    // Missing return type
    testMethod() {
        return "test";
    }
}